src/main/java/org/folio/inventory/consortium/consumers/ConsortiumInstanceSharingHandler.java

```diff
@@ -0,0 +1,273 @@
+        } else {
+          LOGGER.debug("getInstanceById :: Instance with id {} is present on tenant: {}", instanceId, tenantId);
+          promise.complete(success.getResult());
+        }
+      },
+      failure -> {
+        LOGGER.error(format("getInstanceById :: Error retrieving Instance by id %s on tenant %s - %s, status code %s",
+          instanceId, tenantId, failure.getReason(), failure.getStatusCode()));
+        promise.fail(failure.getReason());
+      });
+    return promise.future();
+  }
+
+  private Future<Instance> addInstance(Instance instance, InstanceCollection instanceCollection) {
+    Promise<Instance> promise = Promise.promise();
+    instanceCollection.add(instance, success -> promise.complete(success.getResult()),
+      failure -> {
+        //This is temporary solution (verify by error message). It will be improved via another solution by https://issues.folio.org/browse/RMB-899.
+        if (isNotBlank(failure.getReason()) && failure.getReason().contains(UNIQUE_ID_ERROR_MESSAGE)) {
+          LOGGER.info("addInstance :: Duplicated event received by InstanceId: {}. Ignoring...", instance.getId());
+          promise.fail(new DuplicateEventException(format("Duplicated event by Instance id: %s", instance.getId())));
+        } else {
+          LOGGER.error(format("addInstance :: Error posting Instance %s cause %s, status code %s", instance.getId(), failure.getReason(), failure.getStatusCode()));
+          promise.fail(failure.getReason());
+        }
+      });
+    return promise.future();
+  }
+
+  private Future<Instance> updateInstanceInStorage(Instance instance, InstanceCollection instanceCollection) {
+    Promise<Instance> promise = Promise.promise();
+    instanceCollection.update(instance, success -> promise.complete(instance),
+      failure -> {
+        if (failure.getStatusCode() == HttpStatus.SC_CONFLICT) {
+          promise.fail(new OptimisticLockingException(failure.getReason()));
+        } else {
+          LOGGER.error(format("Error updating Instance - %s, status code %s", failure.getReason(), failure.getStatusCode()));
+          promise.fail(failure.getReason());
+        }
+      });
+    return promise.future();
+  }
+
+  private void sendCompleteEventToKafka(String tenantId, SharingInstance sharingInstance,
+                                        KafkaConfig kafkaConfig, List<KafkaHeader> kafkaHeaders) {
+    sendEventToKafka(tenantId, sharingInstance, COMPLETE, null, kafkaConfig, kafkaHeaders);
+  }
+
+  private void sendEventToKafka(String tenantId, SharingInstance sharingInstance, ConsortiumEnumStatus status, String message,
+                                KafkaConfig kafkaConfig, List<KafkaHeader> kafkaHeaders) {
+
+    ConsortiumEvenType evenType = ConsortiumEvenType.CONSORTIUM_INSTANCE_SHARING_COMPLETE;
+
+    try {
+      LOGGER.info("sendEventToKafka :: tenantId: {}, instanceId: {}, status: {}{}",
```

Comment:
I think there should be a space between the last two {}

Reply:
Fixed.